core/types/rollup_l1_cost.go

```diff
@@ -53,31 +59,55 @@ var (
 
 var L1BlockAddr = common.HexToAddress("0x4200000000000000000000000000000000000015")
 
-// NewL1CostFunc returns a function used for calculating L1 fee cost.
-// This depends on the oracles because gas costs can change over time.
-// It returns nil if there is no applicable cost function.
-func NewL1CostFunc(config *params.ChainConfig, statedb StateGetter) L1CostFunc {
-	cacheBlockNum := ^uint64(0)
```

Comment:
What happened to the cache?

Reply:
Caching of the compute intensive part is performed in Transaction.L1CostData() which stores the result in the internal atomic.

Reply:
Here it's the serialization of the transaction in order to count zeros / non-zeroes. I moved the DB access outside the cost function call and into the initializer so that no longer needs to be cached to keep it from happening repeatedly.